druid/src/main/java/org/apache/calcite/adapter/druid/DruidRules.java

```diff
@@ -205,12 +205,12 @@ public void onMatch(RelOptRuleCall call) {
         intervals = DruidDateTimeUtils.createInterval(
             query.getRowType().getFieldList().get(timestampFieldIdx).getType(),
             RexUtil.composeConjunction(rexBuilder, triple.getLeft(), false));
-        if (intervals == null) {
-          // We can't push anything useful to Druid.
-          residualPreds.addAll(triple.getLeft());
+        if (intervals == null || intervals.isEmpty()) {
+          // Case we have an filter with extract that can not be written as interval push down
+          triple.getMiddle().addAll(triple.getLeft());
         }
       }
-      if (intervals == null && triple.getMiddle().isEmpty()) {
+      if ((intervals == null || intervals.isEmpty()) && triple.getMiddle().isEmpty()) {
```

Comment:
This condition will never be true: if intervals are null or empty, we will have put the corresponding predicates in triple.getMiddle. If they are empty because triple.getLeft was empty and triple.getMiddle is empty too, we will have entered in L198.